packages/sdk/src/modules/Vaults.ts

```diff
@@ -0,0 +1,225 @@
+import { FundOperationMode, SupportedChains, VaultData } from "@midas-capital/types";
+import { BigNumber, constants, Contract, ContractTransaction, utils } from "ethers";
+
+import EIP20InterfaceABI from "../../abis/EIP20Interface";
+import OptimizedVaultsRegistryABI from "../../abis/OptimizedVaultsRegistry";
+import { OptimizedVaultsRegistry } from "../../typechain/OptimizedVaultsRegistry";
+import { getContract } from "../MidasSdk/utils";
+
+import { CreateContractsModule } from "./CreateContracts";
+import { ChainSupportedAssets } from "./FusePools";
+
+export function withVaults<TBase extends CreateContractsModule = CreateContractsModule>(Base: TBase) {
+  const DECIMAL_OFFSET = 9;
+  return class Vaults extends Base {
+    async getAllVaults(): Promise<VaultData[]> {
+      if (this.chainId === SupportedChains.chapel) {
+        try {
+          const optimizedVaultsRegistry = new Contract(
+            this.chainDeployment.OptimizedVaultsRegistry.address,
+            OptimizedVaultsRegistryABI,
+            this.provider
+          ) as OptimizedVaultsRegistry;
+
+          const vaults = await optimizedVaultsRegistry.callStatic.getAllVaults();
+
+          return await Promise.all(
+            vaults.map(async (vault) => {
+              const optimizedAPRVault = this.createOptimizedAPRVault(vault);
+              const mpo = this.createMasterPriceOracle();
+
+              const [
+                asset,
+                totalSupply,
+                supplyApy,
+                adapterCount,
+                emergencyExit,
+                {
+                  performance: performanceFee,
+                  deposit: depositFee,
+                  withdrawal: withdrawalFee,
+                  management: managementFee,
+                },
+              ] = await Promise.all([
+                optimizedAPRVault.callStatic.asset(),
+                optimizedAPRVault.callStatic.estimatedTotalAssets(),
+                optimizedAPRVault.callStatic.supplyAPY(0),
+                optimizedAPRVault.callStatic.adapterCount(),
+                optimizedAPRVault.callStatic.emergencyExit(),
+                optimizedAPRVault.callStatic.fees(),
+              ]);
```

Comment:
these are too many separate RPC calls, let's implement a lens function for these